lib/galaxy/managers/job_connections.py

```diff
@@ -75,7 +76,12 @@ def outputs_derived_from_input_hda(self, input_hda_id):
                 model.JobToInputDatasetAssociation,
                 model.JobToInputDatasetAssociation.job_id == model.JobToOutputDatasetAssociation.job_id,
             )
-            .where(model.JobToInputDatasetAssociation.dataset_id == input_hda_id)
+            .where(
+                expression.and_(
+                    model.JobToInputDatasetAssociation.dataset_id is not None,
+                    model.JobToInputDatasetAssociation.dataset_id == input_hda_id,
```

Comment:
that doesn't look quite right, `input_hda_id` shouldn't be None in the first place

Reply:
Ahhh you're right, I think the change should be to
`model.JobToOutputDatasetAssociation.dataset_id is not None`
instead of
`model.JobToInputDatasetAssociation.dataset_id is not None`

Reply:
`is_not(None)` actually worked, this should be good now!